### PR TITLE
CON-2508- Stale entry check before starting the CSI driver

### PR DESCRIPTION
**Pull request by AnushaY1916**
- Added a new flag called "DIABLE_NODE_PANIC"
- Stale entry check using rescan every time the CSI driver is started

### CODE REVIEW

File: cmd/csi-driver/csi-driver.go [hunk 1]
```diff
@@ -51,6 +52,18 @@ var (
 			} else {
 				log.InitLogging(csiControllerLogFile, nil, true)
 			}
+			// Scanning for stale entries on each node where CSI resides, 
+			// this check makes sure that it runs only on hpe-csi-nodes pods 
+			podName := os.Getenv("POD_NAME")
+			if strings.Contains(podName, "hpe-csi-node-") { //&& !strings.Contains(podName, "hpe-csi-controller-") {
```
Comment (datamattsson) [inline cmd/csi-driver/csi-driver.go:58]: You should be able to check the arguments if the driver is running in node or controller mode (`--node-service`)

File: cmd/csi-driver/csi-driver.go [hunk 2]
```diff
@@ -51,6 +52,18 @@ var (
...
+        	                log.Info("********Cleaning the stale entries before starting the HPE CSI driver********")
+				log.Info("Running rescan-scsi-bus.sh script to flush the stale multipath devices...")
+				output, _, err := util.ExecCommandOutputWithTimeout("rescan-scsi-bus.sh", []string{"-f", "-m", "-r"}, 120)
+				if err != nil {
+					log.Error("unable to flush the stale multipath devices using rescan-scsi-bus.sh script")
+				}
```
Comment (datamattsson) [inline cmd/csi-driver/csi-driver.go:64]: If this fails, we need to understand why and determine if it's fatal. And if it succeeds, we need to ensure the node is still not unhealthy. If any of those two conditions are met, reboot.

**Review by datamattsson — CHANGES_REQUESTED** [review]

File: cmd/csi-driver/csi-driver.go (hunk 1 above)
Comment (AnushaY1916) [reply cmd/csi-driver/csi-driver.go]: Updated

**Comment by AnushaY1916** [comment]
https://github.com/hpe-storage/co-deployments/pull/352 

DISABLE_NODE_PANIC flag handling in co-deployments.

**Pull request closed without merging**